db/change_listener.go

```diff
@@ -270,6 +269,7 @@ func (listener *changeListener) StartNotifierBroadcaster(ctx context.Context) {
 			select {
 			case <-terminator:
 				ticker.Stop()
+				close(listener.broadcastChangesDoneChan)
```

Comment:
I'd keep the defer and move it to the top of the goroutine function personally.

We don't know if/when we'll introduce another case in here that can cause the goroutine to exit (e.g. let's say we started using `context` to close this in addition to the terminator channel.